client/web/src/cody/management/subscription/new/NewCodyProSubscriptionPage.tsx

```diff
@@ -67,25 +58,20 @@ const AuthenticatedNewCodyProSubscriptionPage: FunctionComponent<NewCodyProSubsc
     }, [telemetryRecorder])
 
     useEffect(() => {
-        if (userCodyPlanError) {
-            logger.error('Failed to fetch subscription data', userCodyPlanError)
-        }
-    }, [userCodyPlanError])
-    useEffect(() => {
-        if (subscriptionQueryResult.isError) {
+        if (subscriptionQueryResult.error) {
             logger.error('Failed to fetch subscription data', subscriptionQueryResult.error)
         }
-    }, [subscriptionQueryResult.isError, subscriptionQueryResult.error])
-
-    if (userCodyPlanLoading || subscriptionQueryResult.isLoading) {
-        return <LoadingSpinner className="mx-auto" />
-    }
+    }, [subscriptionQueryResult.error])
 
     // If the user already has a Cody Pro subscription, direct them back to the Cody Management page.
-    if (!addSeats && userCodyPlanData?.currentUser?.codySubscription?.plan === CodySubscriptionPlan.PRO) {
+    if (!addSeats && codySubscription.plan === CodySubscriptionPlan.PRO) {
         return <Navigate to={CodyProRoutes.Manage} replace={true} />
     }
 
+    if (subscriptionQueryResult.isLoading) {
+        return <LoadingSpinner className="mx-auto" />
+    }
```

Comment:
Nit: a comment like this could be helpful
```suggestion
    // Display spinner without page header because without the subscription, we don't know which header to show
    if (subscriptionQueryResult.isLoading) {
        return <LoadingSpinner className="mx-auto" />
    }
```